fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -509,6 +510,7 @@ enum TypeCode {
         INT(Integer.class, FieldDescriptorProto.Type.TYPE_INT32, true, true),
         LONG(Long.class, FieldDescriptorProto.Type.TYPE_INT64, true, true),
         STRING(String.class, FieldDescriptorProto.Type.TYPE_STRING, true, false),
+        VERSION(FDBRecordVersion.class, FieldDescriptorProto.Type.TYPE_BYTES, true, false),
```

Comment:
What happens if you try to put a `VERSION` into a `RECORD`? Is the normal _primitive type_ logic sufficient to deal with this new type so it can figure out how to create the photo descriptors, etc.? Are there tests for this?

Reply:
Another design question: is it possible to avoid introducing the `VERSION` at all as a primitive type? it seems to me that `version` is merely a function produces a `byte[]` array.

Can't we model it as such, e.g. to make `VersionValue` return a `byte[]`?